### PR TITLE
[BD-46] docs: update spacings table using dataTable

### LINKED ISSUE #2539: [alpha] Documentation spacing table styles
## Bugs

The table that shows the spacing utility classes don't have all the styles, making the information difficult to read.

image

**Expected behavior**

Show the table with the correct spacing and borders.

image

### PR DESCRIPTION
**Pull request by khudym**
## Description
Update All Spacing Utility Classes table using DataTable component
[Issue](https://github.com/openedx/paragon/issues/2539)

### Deploy Preview

[Deploy](https://deploy-preview-2677--paragon-openedx.netlify.app/foundations/spacing)

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Were your changes tested in the `example` app?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @khudym!


When this pull request is ready, tag your edX technical lead.

**Comment by viktorrusakov** [comment]
@khudym Please fix lint issues, the pipeline is failing right now.

File: www/src/pages/foundations/spacing.tsx [hunk 1]
```diff
@@ -95,6 +95,17 @@ export default function SpacingPage() {
     pixelValue: <PixelCell spacer={value} />,
   }));
 
+  const allUtilityClassesTabel= (prefix) => sizes.map(size => {
```
Comment (PKulkoRaccoonGang) [inline www/src/pages/foundations/spacing.tsx:98]: It seems to me that if it is a function, then it should be a verb. As an option: `createUtilityClassesTabel`.
What do you think?

**Review by PKulkoRaccoonGang — COMMENTED** [review]
@khudym Now in the current version of the [Paragon documentation site](https://paragon-openedx.netlify.app/foundations/spacing#all-spacing-utility-classes), we have a table where each cell element is located in a `<code>` tag. Let's try to repeat the same visual display in the new table for All Spacing Utility Classes.
Let's also add an indent before the "Padding" heading
image

**Review by viktorrusakov — APPROVED** [review]

**Pull request merged**